service/actions/websocket.js

```diff
@@ -224,7 +224,7 @@ async function handleNewPostEvent(msg, dispatch, getState) {
         }
     ]), getState);
 
-    if (userId === users.currentId || post.channel_id === currentChannelId || isSystemMessage(post)) {
+    if ((userId === users.currentId && !isSystemMessage(post)) || post.channel_id === currentChannelId || shouldIgnorePost(post)) {
```

Comment:
And this behavior might further differ from what's expected:
If we receive a non-ignored system message for an action done by the current user (e.g. they updated the channel purpose) then it will mark that channel as unread for them. We probably don't want to mark it unread if they're the one that triggered the system message.